### PR TITLE
Add support for loadbalancer stats

### PR DESCRIPTION
**Pull request by jlongstaf**
@richbrowne, @mattgreene 
#### What issues does this address?
WIP #40

#### What's this change do?
Implements support for lbaas-loadbalancer-stats command.

#### Where should the reviewer start?
Start with agent_manager, collect_stats(), then icontrol_driver get_stats()

#### Any background context?
NOTE: See original v1 implementation of get_stats in icontrol_driver. The v1 code has additional work to update member status. I didn't implement this as it is independent of loadbalancer stats. If we still want this, it should be added under a separate issue.

### CODE REVIEW

File: f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py [hunk 1]
```diff
@@ -826,105 +824,27 @@ def delete_health_monitor(self, health_monitor, service):
...
+        stats = ['serverside.bitsIn',
+                 'serverside.bitsOut',
+                 'serverside.curConns',
+                 'serverside.totConns']
+
+        try:
+            # sum pool stats for all BIG-IPs
+            pool_stats = self.lbaas_builder.get_pool_stats(service, stats)
+
+            # convert to bytes
```
Comment (mattgreene) [inline f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py]: Use the pre-defined values from lb_const, such as lb_const.STATS_IN_BYTES

File: f5_openstack_agent/lbaasv2/drivers/bigip/pool_service.py [hunk 2]
```diff
@@ -185,3 +185,35 @@ def _get_monitor_helper(self, service):
...
+        :return: A dict with key/value pairs for each stat defined in
+        input stats.
+        """
+        collected_stats = {}
+        for stat in stats:
+            collected_stats[stat] = 0
+
+        pool = self.service_adapter.get_pool(service)
+        part = pool["partition"]
+        for bigip in bigips:
```
Comment (mattgreene) [inline f5_openstack_agent/lbaasv2/drivers/bigip/pool_service.py]: Can a pool span multiple BIGIPs?  Maybe check if the pool exists on the bigip before calling get_stats.

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_helper.py [hunk 3]
```diff
@@ -246,3 +246,26 @@ def _collection(self, bigip):
                       "resource %s", self.resource_type)
             raise KeyError("No collection available for %s" %
                            (self.resource_type))
+
+    def get_stats(self, bigip,  name=None, partition=None, stats=None):
```
Comment (mattgreene) [inline f5_openstack_agent/lbaasv2/drivers/bigip/resource_helper.py]: stats must default to [] not None.  Line 266 will fail with the following:

>>> stats = None
>>> for stat in stats:
...   print stat
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable

**Comment by mattgreene** [comment]
Done with review.  Ok to merge after resolving comments, and feedback from Rich.

File: f5_openstack_agent/lbaasv2/drivers/bigip/pool_service.py (hunk 2 above)
Comment (jlongstaf) [inline f5_openstack_agent/lbaasv2/drivers/bigip/pool_service.py]: Added check if resource exists() in get_stats() in resource_helper.py

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_helper.py (hunk 3 above)
Comment (jlongstaf) [inline f5_openstack_agent/lbaasv2/drivers/bigip/resource_helper.py]: Fixed to default to [].

File: f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py (hunk 1 above)
Comment (jlongstaf) [inline f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py]: Fixed to include lb_const values.

**Pull request merged**